### ISSUE TITLE
Unrecognized option 'extensions'

### ISSUE THREAD

**Issue by FractalHQ** [dominikg/svite#5]
This is a screenshot from index.svelte from a (relatively) unmodified version of the [routify-mdsvex example](https://github.com/dominikg/svite/tree/master/examples/routify-mdsvex). I'm getting this error from every file I access.

![https://user-images.githubusercontent.com/55504972/86011211-6d921f00-b9ea-11ea-9116-81abbad9634b.png](https://user-images.githubusercontent.com/55504972/86011211-6d921f00-b9ea-11ea-9116-81abbad9634b.png)

I'm getting it from the [official Svelte plugin](https://github.com/sveltejs/language-tools.git), and the [unofficial one](https://github.com/UnwrittenFun/svelte-vscode.git).

It will periodically fix itself, but I can't reliably reproduce the conditions in which it works.

I tried passing the svelte config through the vite config, but it didn't seem to help.  Sorry if I'm missing something obvious!

PS:  Thank you so much for Svite, it's amazing and is the future.  I think I prefer it to snowpack / nollup.

**Comment by dominikg** [comment]
This is properbly caused by the language-tools and svite expecting different content in svite.config.js

language-tools  uses it as [CompileOptions](https://github.com/sveltejs/svelte/blob/5486d67adf2d48820e94a1812028dba9392975b7/src/compiler/interfaces.ts#L107) for the svelte compiler. 

svite uses it as rollup-plugin-svelte [options](https://github.com/sveltejs/rollup-plugin-svelte/blob/907d7137564273a2babaf058f59477c41e1a8168/index.d.ts#L25). (note, the interface is incomplete, see [usage](https://github.com/sveltejs/rollup-plugin-svelte#usage) for more

I've got to think about how to resolve this conflict, but as a workaround you can specify options only available for rollup-plugin-svelte via 
```js
//vite.config.js
const svite = require('svite');
module.exports = {
  plugins: [svite({
    svelte: {
      extensions:['.svelte','.svx']
    }
  })],
};
```

svite merges config in svelte.config.js with sviteOptions.svelte, that way you can hide the unrecognized option from language-tools

**Comment by FractalHQ** [comment]
That seemed to have fixed the issue!  Unfortunately, saving a file now gives me this error (although its gone once I refresh the page)
<img width="875" alt="Screenshot" src="https://user-images.githubusercontent.com/55504972/86198259-31afa480-bb25-11ea-95d9-87b2c458e6f5.png">
I'll see what I can do with some fiddling.

**Comment by dominikg** [comment]
This seems to be caused when trying to apply an update to a module that does not have any state.

You can test this if you add some state like foo:
```html
<script>
let foo=0;
</script>
```
all updates work. once you remove the state, the update that removes still works, but the next update fails. 

There is a newer version of svelte-hmr, i'm checking if that fixes the issue.

**Comment by FractalHQ** [comment]
Ah!  Thanks for the insight- this is a relatively painless work-around.

**Comment by dominikg** [comment]
i just published 0.1.0 with updated svelte-hmr and this should fix your issue.

**Comment by dominikg** [comment]
the [workaround mentioned above](https://github.com/dominikg/svite/issues/5#issuecomment-651910083) should no longer be needed after this PR https://github.com/sveltejs/language-tools/pull/270 has been released.

I'm closing this issue as there is an easy workaround and a proper solution is imminent. If there are any questions left feel free to hunt me down.
